### PR TITLE
Revamping the scripts used to start/stop Mycroft

### PR DESCRIPTION
**Pull request by penrods**
Reworking the mechanisms used to interact with Mycroft.  The old
mycroft.sh utilized the 'screen' command, but now with the CLI it
it easier to use that for viewing log files.  Plus there was
confusion between when to use start.sh and when to use mycroft.sh.

Now things are wrapped in the simple and easily discoverable scripts:
* start-mycroft.sh
* stop-mycroft.sh

The start-mycroft.sh allows you to start the background services and
to invoke the various tools (e.g. the CLI, unit tests, etc).

==== Fixed Issues ====
#1107 - Revamping scripts used to start/stop Mycroft

====  Documentation Notes ====
This will require revamp of the getting started guides, etc.

### CODE REVIEW

**Comment by KathyReid** [comment]
## Comments as the voice of the developer

### Making starting and stopping services easier and more intuitive
I am in favour of this change because it eliminates the dependency on `screen` and makes it more intuitive to start and stop Mycroft services, and removes developer confusion around mycroft.sh, mycroft-start.sh and mycroft-stop.sh

### Documentation dependencies
The Mycroft documentation is fragmented into multiple locations. Successful release of this commit relies on identifying where the documentation references the old shell scripts, and updating the documentation. This is a large piece of work on its own. 

_For discussion_ What is the impact of releasing this change before updating the documentation - will it cause more confusion for developers?

### Recommendation

* To first identify documentation dependencies
* Release the change
* Update documentation soon after change

**Comment by penrods** [comment]
I _think_ most devs would figure out how this works despite documentation mis-matches.  I agree that I'd like to have the change identified beforehand, but even if there is a period of incorrect documentation, I don't think most will be that lost.  I'm going to push for more discussion later today, as many involved with the project really spin-up over the weekend.

**Comment by skeledrew** [comment]
I endorse this move to cut down on the number of scripts. I'd actually go even further and ask for a single source of truth. So a mycroft.sh script that can do everything from install to start/stop, just like git and many other commands work. Running without a subcommand would start a default instance (maybe there can be multiple controlled from the same place?) and ask a user to invoke install/fix subcommand if it doesn't seem to be installed properly, enter debug mode, etc. If a dev wants to add extra scripts to call the particular subcommands then they're welcome.

**Comment by JarbasAI** [comment]
`usage: ${script} [command]` shouldnt this be `usage: ${script} [command] [params]`

for things like `mycroft-start.sh cli simple`

File: stop-mycroft.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    end-process skills
+    ;;
+  "voice")
+    end-process speech
+    ;;
+
+
+  *)
+    help
+    exit_code=0
```
[the remark this answers is not in the document]
Comment (el-tocino) [reply stop-mycroft.sh]: exit_code should be removed.  Help function includes "exit 1" already.

**Comment by penrods** [comment]
All fixed, I'm doing it!

**Pull request merged**